misk-redis/src/main/kotlin/misk/redis/RealRedis.kt

```diff
@@ -232,13 +232,25 @@ class RealRedis(private val jedisPool: JedisPool) : Redis {
 
   override fun lpop(key: String, count: Int): List<ByteString?> {
     return jedisPool.resource.use { jedis ->
-      jedis.lpop(key.toByteArray(charset), count).map { it?.toByteString() }
+      jedis.lpop(key.toByteArray(charset), count)?.map { it?.toByteString() } ?: emptyList()
+    }
+  }
+
+  override fun lpop(key: String): ByteString? {
+    return jedisPool.resource.use { jedis ->
+      jedis.lpop(key.toByteArray(charset))?.toByteString()
     }
   }
 
   override fun rpop(key: String, count: Int): List<ByteString?> {
     return jedisPool.resource.use { jedis ->
-      jedis.rpop(key.toByteArray(charset), count).map { it?.toByteString() }
+      jedis.rpop(key.toByteArray(charset), count)?.map { it?.toByteString() } ?: emptyList()
```

Comment:
jedis can return a null list

Reply:
Will rpop with a count still throw the same error? Looks like it's using the same API? Do I need to switch to use the one without count?

Reply:
Yeah since we're working off a v5.0 instance, we'll need to use the rpop without count.

I _guess_ I could change the underlying impl to use the rpop without count, but I don't want to forgo efficiency for clients of newer Redis.